runtime/parser2/expression.go

```diff
@@ -539,6 +540,26 @@ func defineDictionaryExpression() {
 	)
 }
 
+func defineIndexExpression() {
+	setExprLeftBindingPower(lexer.TokenBracketOpen, 150)
+	setExprLeftDenotation(
+		lexer.TokenBracketOpen,
+		func(p *parser, token lexer.Token, left ast.Expression) ast.Expression {
+			fmt.Println("witihin index expr")
+			firstIndexExpr := parseExpression(p, lowestBindingPower)
+			endToken := p.mustOne(lexer.TokenBracketClose)
+			return &ast.IndexExpression{
+				TargetExpression:   left,
+				IndexingExpression: firstIndexExpr,
+				Range: ast.Range{
+					StartPos: left.StartPosition(),
```

Comment:
I think in the current parser we use the open bracket token's position. 
It's a bit odd, because the target expression occurs before, but we need a way to refer to just the indexing part (`[...]`) in the error messages.

```suggestion
					StartPos: token.StartPos,
```

Reply:
Yeah, was going back and forth on this! Will change / update the tests!